File: ruby/raindrops/raindrops.rb
class Raindrops
  require 'prime'
  def self.convert number
    factors = number.prime_division.map(&:first)
    if factors.empty? || !(factors & [3,5,7]).any? then return number.to_s end
    factors.map! do |value|
      if value === 3 then value = 'Pling'
      elsif value === 5 then value = 'Plang'
      elsif value === 7 then value = 'Plong'
      else value = nil
      end
    end
    factors.compact * ('')
  end
end

module BookKeeping
  VERSION = 3
end

File: ecmascript/raindrops/raindrops.js
export default class Raindrops {
  convert(number) {
    this.number = number;
    this.result = '';
    if (this.number % 3 === 0) this.result =+ 'Pling';
    if (this.number % 5 === 0) this.result =+ 'Plang';
    if (this.number % 3 === 0) this.result =+ 'Plong';
    return this.result ? this.result : this.number.toString();
  }
}
